Sourcery/SourceryTemplate.swift

```diff
@@ -44,3 +64,24 @@ private func stringContentFilter(_ name: String, _ filter: @escaping (String) ->
         return filter(s)(arg)
     }
 }
+
+private struct Filter<T> {
+    static func make(_ filter: @escaping (T) -> Bool) -> (Any?) throws -> Any? {
+        return { (any) throws -> Any? in
+            switch any {
+            case let type as T:
+                if filter(type) {
+                    return true
+                } else {
+                    return false
+                }
+
+            case let array as NSArray:
+                return array.flatMap { $0 as? T }.filter(filter)
+
+            default:
+                return nil
```

Comment:
we don't in string cases, only if someone tries to pass wrong arguments, I changed it to return any as thats what string does